pkg/crmanager/worker.go

```diff
@@ -622,13 +619,19 @@ func (crMgr *CRManager) syncVirtualServers(
 			)
 		}
 
+		if isVSDeleted {
+			crMgr.handleVSDeleteForDataGroups(virtual, rsName)
+		}
+
 		// Delete rsCfg if no corresponding virtuals exist
 		// Delete rsCfg if it is HTTP rsCfg and the CR VirtualServer does not handle HTTPTraffic
 		if (len(virtuals) == 0) ||
 			(portStruct.protocol == "http" && !doesVSHandleHTTP(virtual)) {
-			crMgr.resources.deleteVirtualServer(rsName)
-			//TODO: what if last VirtualServer does not have cidr
-			crMgr.releaseIP(virtual.Spec.Cidr, virtual.Spec.Host)
+			crMgr.deleteVirtualServer(rsName)
+			if len(virtuals) == 0 && crMgr.ipamCli != nil {
```

Comment:
@subbuv26 These changes are already incorporated into another PR.

Reply:
That is good. I will remove this line and update PR.